gef.py

```diff
@@ -7733,18 +7733,26 @@ def reason():
             err("No thread selected")
             return
 
+        selected_thread = gdb.selected_thread()
+
         for i, thread in enumerate(threads):
-            line = """[{:s}] Id {:d}, Name: "{:s}", """.format(Color.colorify("#{:d}".format(i), "bold pink"),
-                                                               thread.num, thread.name or "")
+            line = """[{:s}] Id {:d}, """.format(Color.colorify("#{:d}".format(i), "bold green" if thread==selected_thread  else "bold pink"), thread.num)
+            if thread.name:
+                line += """Name: "{:s}", """.format(thread.name)
             if thread.is_running():
                 line += Color.colorify("running", "bold green")
             elif thread.is_stopped():
                 line += Color.colorify("stopped", "bold red")
+                thread.switch()
```

Comment:
Why are we switching to the thread? Is this needed to get the frame?

Reply:
If I understand [this](https://www-zeuthen.desy.de/unix/unixguide/infohtml/gdb/Frames-In-Python.html) right, we can get frames only for selected thread